### PR TITLE
Giddh daterange picker

### PR DESCRIPTION
**Pull request by shalinee-walkover**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

**Review by ravinderwalkover — APPROVED** [review]

File: apps/web-giddh/src/app/services/general.service.ts [hunk 1]
```diff
@@ -62,7 +62,7 @@ export class GeneralService {
             moment().subtract(1, 'year').endOf('year')
         ]
     };
-
+    public _selectedRange: Subject<any> = new Subject();
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/services/general.service.ts:65]: Avoid using underscores
Do add JS doc

File: apps/web-giddh/src/app/services/general.service.ts [hunk 2]
```diff
@@ -114,6 +114,14 @@ export class GeneralService {
         this._createNewCompany = newCompanyRequest;
     }
 
+    public selectRange(dates: any) {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/services/general.service.ts:117]: Add JS doc and return type

File: apps/web-giddh/src/app/services/general.service.ts [hunk 3]
```diff
@@ -114,6 +114,14 @@ export class GeneralService {
         this._createNewCompany = newCompanyRequest;
     }
 
+    public selectRange(dates: any) { 
+        this._selectedRange.next(dates);
+    }
+
+    get selectedRange(): any {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/services/general.service.ts:121]: Add JS doc

File: apps/web-giddh/src/app/theme/giddh-daterangepicker/giddh-daterangepicker.component.ts [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+    public giddhDaterangepickerSidebar = GiddhDaterangepickerSidebarComponent;
+
+    /** Emitting selected date object as output */
+    @Output() public dateSelected: EventEmitter<any> = new EventEmitter<any>();
+    /** Taking start date */
+    @Input() public inputStartDate: any = '';
+    /** Taking end date */
+    @Input() public inputEndDate: any = '';
+
+    @ViewChild('picker', {static: true}) picker: MatDatepicker<Date>;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/giddh-daterangepicker/giddh-daterangepicker.component.ts:33]: Add JS doc for all new members

File: apps/web-giddh/src/app/theme/giddh-daterangepicker/giddh-daterangepicker.component.ts [hunk 5]
```diff
@@ -0,0 +1,288 @@
...
+     * Updates the value on value change event
+     *
+     * @param {SimpleChanges} changes
+     * @memberof GiddhDaterangepickerComponent
+     */
+    public ngOnChanges(changes: SimpleChanges): void {
+        if (changes.inputStartDate && changes.inputStartDate.currentValue) {
+            this.startDate = changes.inputStartDate.currentValue.toDate();
+        }
+        if (changes.inputStartDate && changes.inputEndDate.currentValue) {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/giddh-daterangepicker/giddh-daterangepicker.component.ts:99]: Check should be on `inputEndDate`

File: apps/web-giddh/src/app/theme/giddh-daterangepicker/giddh-daterangepicker.component.ts [hunk 6]
```diff
@@ -0,0 +1,288 @@
...
+            this.endDate = changes.inputEndDate.currentValue.toDate();
+        }
+    }
+
+    public ngOnDestroy(): void {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/giddh-daterangepicker/giddh-daterangepicker.component.ts:104]: Add JS doc

File: apps/web-giddh/src/app/theme/giddh-daterangepicker/giddh-daterangepicker.component.ts [hunk 7]
```diff
@@ -0,0 +1,288 @@
...
+    public dateChange(type: string, event: MatDatepickerInputEvent<Date>): void {
+        if(type === "start") {
+            this.startDate = moment(event.value, GIDDH_DATE_FORMAT).toDate();
+        }
+        if(type === "end") {
+            this.endDate = moment(event.value, GIDDH_DATE_FORMAT).toDate();
+        }
+    }
+
+    public openDatepicker(): void {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/giddh-daterangepicker/giddh-daterangepicker.component.ts:124]: Add JS doc

File: apps/web-giddh/src/app/theme/giddh-daterangepicker/giddh-daterangepicker.component.ts [hunk 8]
```diff
@@ -0,0 +1,288 @@
...
+/** Custom sidebar component for datepicker. */
+@Component({
+    selector: 'giddh-daterangepicker-sidebar',
+    templateUrl: './giddh-daterangepicker-sidebar.component.html',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+
+export class GiddhDaterangepickerSidebarComponent<D> implements OnInit, OnDestroy {
+    /* This will store available date ranges */
+    public datePickerRanges: any = GIDDH_DATE_RANGE_PICKER_RANGES;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/giddh-daterangepicker/giddh-daterangepicker.component.ts:140]: Add JS doc to all members and methods

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss [hunk 9]
```diff
@@ -1424,41 +1450,82 @@ td span {
...
+                                box-shadow: none;
+                            }
+                        }
+                        span {
+                            display: flex;
+                            align-items: flex-start;
+                            align-items: center;
+                            color: $gray;
+                        }
+                        &.focusedDate{
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss:1508]: Avoid using camelCase

**Review by prateek-giddh — CHANGES_REQUESTED** [review]

**Review by prateek-giddh — APPROVED** [review]

**Pull request merged**